### PR TITLE
Aufgabe 02 Jan Merz

### PR DESCRIPTION
**Pull request by TrueEsel**

### CODE REVIEW

File: Exercises/Jan Merz/exercises_02.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        try:
+            castValue = args[i]
+            complex(castValue)
+            numberList.append(args[i])
+        except ValueError:
+            pass
+
+        if(len(args[i]) == 1):
+            charList.append(args[i])
+    print(numberList, " are numbers and ", charList, " are single characters")
```
Comment (christianrietzsch) [inline Exercises/Jan Merz/exercises_02.py:78]: the function should return two lists: `return (numberList, charList)`

File: Exercises/Jan Merz/exercises_02.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+#   as a number.
+#   The second list should contain all strings which contain just one character.
+# Think of some good inputs to test this functionality, write down at least three
+# examples and verify that the output for these examples is correct.
+
+def splitArgs(*args):
+    numberList = []
+    charList = []
+
+    for i in range(len(args)):
```
Comment (christianrietzsch) [inline Exercises/Jan Merz/exercises_02.py:68]: you should just directly loop through the elements of args: `for element in args:` if you never use the index
except to reference an element: `args[i]` -> `element`

File: Exercises/Jan Merz/exercises_02.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#    of your working memory, add them, and store the result in a third position of your working memory.
+#    The three numbers immediately after your opcode indicate the memory locations to read (first two values)
+#    and write (third value) respectively. 
+#    After executing the addition you should move to the next opcode by stepping forward 4 positions.
+#  2 indicates multiplication. Otherwise the same rules apply as for opcode 1.
+# 99 indicates halt. the program should stop after encountering the opcode 99.
+# After the program stops, the function should return the value in the first location (address 0) 
+# of your working memory.
+
+def computer_sim(arr):
```
Comment (christianrietzsch) [inline Exercises/Jan Merz/exercises_02.py:17]: good solution

File: Exercises/Jan Merz/exercises_02.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        except ValueError:
+            pass
+
+        if(len(args[i]) == 1):
+            charList.append(args[i])
+    print(numberList, " are numbers and ", charList, " are single characters")
+
+splitArgs("1", "1.47546", "99999999", "Hallo", "h", "1String", "5+5j")
+splitArgs("f", "hr", "1907074939", "59347f92", "446.45", "t", "ab", "3 + 9j")
+splitArgs("0", "-4254", "0.00000000001", "ä", "pß", "?")
```
Comment (christianrietzsch) [inline Exercises/Jan Merz/exercises_02.py:82]: nice test cases including negative numbers, floats and complex (imaginary) numbers

**Review by christianrietzsch — CHANGES_REQUESTED** [review]
please add the required return value in task 2

File: Exercises/Jan Merz/exercises_02.py (hunk 1 above)
Comment (TrueEsel) [reply Exercises/Jan Merz/exercises_02.py:78]: fixed that

**Review by christianrietzsch — APPROVED** [review]

**Pull request merged**